### PR TITLE
Display minimap on landscape screen

### LINKED ISSUE #3137: Display minimap on landscape screen
It can be a good idea to a minimap (like in RTS games) somewhere on the landscape screen to ease navigation in the landscape

### PR DESCRIPTION
**Pull request by abdelfetah18**
Fix #3137

### CODE REVIEW

File: src/main/webapp/app/module/primary/landscape-minimap/LandscapeMiniMap.component.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      setupDimensions();
+
+      landscapeContainer.addEventListener('scroll', trackScroll);
+      window.addEventListener('resize', setupDimensions);
+    });
+
+    const removeDataSelectorAttrs = (data: string): string => {
+      const regex = /data-selector="[^"]*"/g;
+      const result = data.replace(regex, '');
+      return result;
```
Comment (qmonmert) [inline src/main/webapp/app/module/primary/landscape-minimap/LandscapeMiniMap.component.ts:36]: simplify :)
```
const result = data.replace(regex, '');
return result;
``` 
=>
```
return data.replace(regex, '');
```

**Comment by qmonmert** [comment]
@abdelfetah18 can you squash some commits? 🙂

**Comment by abdelfetah18** [comment]
@qmonmert is this good ?

File: src/main/webapp/app/module/primary/landscape-minimap/LandscapeMiniMap.vue [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+<script lang="ts" src="./LandscapeMiniMap.component.ts"></script>
+
+<style>
+#minimap-container {
+  position: fixed;
+  bottom: 20px;
+  left: 20px;
+  min-width: 20px;
+  z-index: 100;
+  background-color: #0000009a;
```
Comment (qmonmert) [inline src/main/webapp/app/module/primary/landscape-minimap/LandscapeMiniMap.vue:12]: it's desired 8 digits?

Comment (abdelfetah18) [reply src/main/webapp/app/module/primary/landscape-minimap/LandscapeMiniMap.vue:12]: 6 digits for the color which is black "000000"
2 digits for the alpha "9A"

Comment (qmonmert) [reply src/main/webapp/app/module/primary/landscape-minimap/LandscapeMiniMap.vue:12]: 👍

**Review by qmonmert — APPROVED** [review]
LGTM

File: src/test/javascript/spec/module/primary/landscape/LandscapeMiniMapComponent.spec.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import { LandscapeMiniMapVue } from '@/module/primary/landscape-minimap';
+import { mount } from '@vue/test-utils';
+import { describe, it, expect, vi } from 'vitest';
+import { wrappedElement } from '../../../WrappedElement';
+import sinon, { SinonStub } from 'sinon';
```
Comment (pascalgrimaud) [inline src/test/javascript/spec/module/primary/landscape/LandscapeMiniMapComponent.spec.ts:5]: can you follow https://github.com/jhipster/jhipster-lite/pull/7505 and replace sinon by vitest here plz?

**Comment by abdelfetah18** [comment]
@pascalgrimaud any other things to change ?

**Comment by pascalgrimaud** [comment]
@abdelfetah18 : I'll try to find time to test it. Thanks for waiting :-)

**Comment by abdelfetah18** [comment]
> @abdelfetah18 : I'll try to find time to test it. Thanks for waiting :-)

@pascalgrimaud : No problem, take your time.

**Review by DamnClin — DISMISSED** [review]
LGTM, some improvements (maybe for later): 

More visible border around current window, may be hard to see;
Change cursor in the movable part to see it is movable;
Hiddeable minimap;
Add interactions (hover, selection, ...) to match current landscape state.

But really good idea to tackle it that way, really liked it!

**Comment by abdelfetah18** [comment]
> More visible border around current window, may be hard to see;

This one, i will do it right now

> Change cursor in the movable part to see it is movable;
> Hiddeable minimap;
> Add interactions (hover, selection, ...) to match current landscape state.

can be done in another PRs.

> But really good idea to tackle it that way, really liked it!

Thank you

**Comment by DamnClin** [comment]
In some cases border height is off, example on screenshot I'm ne seeing the full height: 
Capture d’écran du 2023-09-24 11-28-00

**Comment by abdelfetah18** [comment]
I am using the dark mode, and the border color is white.
i will change the color for the light mode

**Comment by DamnClin** [comment]
> I am using the dark mode, and the border color is white. i will change the color for the light mode

It's not the color, it's the height :D. Here it is displaying full height but my window is too small to display full landscape height

**Comment by abdelfetah18** [comment]
@DamnClin this must fix the issue,
i was calculating the wrong height for the minimap viewer.

**Pull request merged**